components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -268,137 +261,63 @@ func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) adm
-					proxyEnvVars["REQUESTS_CA_BUNDLE"] = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
-					proxyEnvVars["SSL_CERT_FILE"] = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
-				}
-				return true
-			}
-		}
-	}
-	return false
-
-}
-
 // InjectDecoder injects the decoder.
 func (w *NotebookWebhook) InjectDecoder(d *admission.Decoder) error {
 	w.Decoder = d
 	return nil
 }
 
-func InjectProxyConfig(notebook *nbv1.Notebook) error {
-	notebookContainers := &notebook.Spec.Template.Spec.Containers
-	var imgContainer corev1.Container
+func CheckAndMountCACertBundle(ctx context.Context, cli client.Client, notebook *nbv1.Notebook) error {
 
-	// Add trusted-ca volume
-	notebookVolumes := &notebook.Spec.Template.Spec.Volumes
-	certVolumeExists := false
-	certVolume := corev1.Volume{
-		Name: "trusted-ca",
-		VolumeSource: corev1.VolumeSource{
-			ConfigMap: &corev1.ConfigMapVolumeSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: "trusted-ca",
-				},
-				Items: []corev1.KeyToPath{{
-					Key:  "ca-bundle.crt",
-					Path: "tls-ca-bundle.pem",
-				},
-				},
-			},
-		},
-	}
-	for index, volume := range *notebookVolumes {
-		if volume.Name == "trusted-ca" {
-			(*notebookVolumes)[index] = certVolume
-			certVolumeExists = true
-			break
-		}
-	}
-	if !certVolumeExists {
-		*notebookVolumes = append(*notebookVolumes, certVolume)
-	}
+	configMapName := "odh-trusted-ca-bundle"
 
-	// Update Notebook Image container with env variables and Volume Mounts
-	for _, container := range *notebookContainers {
-		// Update notebook image container with env Variables
-		if container.Name == notebook.Name {
-			var newVars []corev1.EnvVar
-			imgContainer = container
-
-			for key, val := range proxyEnvVars {
-				keyExists := false
-				for _, env := range imgContainer.Env {
-					if key == env.Name {
-						keyExists = true
-						// Update if Proxy spec is updated
-						if env.Value != val {
-							env.Value = val
-						}
-					}
-				}
-				if !keyExists {
-					newVars = append(newVars, corev1.EnvVar{Name: key, Value: val})
-				}
-			}
+	// Fetch the list of ConfigMaps from the cluster
+	configMapList := &corev1.ConfigMapList{}
+	if err := cli.List(ctx, configMapList); err != nil {
+		return err
+	}
 
-			// Update container only when required env variables are not present
-			imgContainerExists := false
-			if len(newVars) != 0 {
-				imgContainer.Env = append(imgContainer.Env, newVars...)
-			}
+	// Search for the odh-trusted-ca-bundle ConfigMap
+	for i := range configMapList.Items {
+		cm := &configMapList.Items[i]
+		if cm.Name == configMapName {
 
-			// Create Volume mount
-			volumeMountExists := false
-			containerVolMounts := &imgContainer.VolumeMounts
-			trustedCAVolMount := corev1.VolumeMount{
-				Name:      "trusted-ca",
+			volumeName := "trusted-ca"
+			volumeMountPath := "/etc/pki/ca-trust/extracted/pem"
+			volumeMount := corev1.VolumeMount{
+				Name:      volumeName,
+				MountPath: volumeMountPath,
```

Comment:
```suggestion
				MountPath: volumeMountPath,
				subPath: odh-trusted-ca-bundle.crt
```
we should use the subpath , so we don't override any existing bits

Reply:
Check out the print screen bellow is an evaluation of the SubPath 
![image](https://github.com/opendatahub-io/kubeflow/assets/42587738/7ec4e5e7-8486-489c-bdfd-8f160a25fc88)